### PR TITLE
HDDS-8219. Run HA secure tests from s3g container

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

In `ozonesecure-ha` environment SCM containers have hostnames like `scm1.org`, but all Kerberos principals for SCM nodes are in the form `<username>/scm@EXAMPLE.COM`.  Since no host named `scm` exists, lookup takes quite some time:

```
$ OZONE_LOGLEVEL=TRACE ozone sh volume create /vol1 |& grep 'lookup'
[main] TRACE security.SecurityUtil: Name lookup for om2 took 1 ms.
[main] TRACE security.SecurityUtil: Name lookup for om3 took 0 ms.
[main] TRACE security.SecurityUtil: Name lookup for om1 took 0 ms.
[main] TRACE security.SecurityUtil: Name lookup for scm took 497 ms.
```

We can improve this simply by executing the same commands from another container, e.g. `s3g`:

```
$ OZONE_LOGLEVEL=TRACE ozone sh volume create /vol2 |& grep 'lookup'
[main] TRACE security.SecurityUtil: Name lookup for om2 took 1 ms.
[main] TRACE security.SecurityUtil: Name lookup for om3 took 0 ms.
[main] TRACE security.SecurityUtil: Name lookup for om1 took 0 ms.
[main] TRACE security.SecurityUtil: Name lookup for s3g took 0 ms.
```

Tests for bucket links need a minor change to remove hard-coded `scm` from permissions in test setup.

~Opening as draft, since this is based on HDDS-7975 (#4437).~

https://issues.apache.org/jira/browse/HDDS-8219

## How was this patch tested?

Regular CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/4488033748

### CODE REVIEW

File: hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config [hunk 1]
```diff
@@ -47,6 +48,7 @@ OZONE-SITE.XML_ozone.recon.db.dir=/data/metadata/recon
 OZONE-SITE.XML_ozone.recon.address=recon:9891
 OZONE-SITE.XML_ozone.recon.http-address=0.0.0.0:9888
 OZONE-SITE.XML_ozone.recon.https-address=0.0.0.0:9889
+OZONE-SITE.XML_dfs.container.ratis.datastream.enabled=true
```
Comment (kerneltime) [inline hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config:51]: It this needed for this PR?

Comment (adoroszlai) [reply hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config:51]: Yes, Ozone FS tests have a test case for write with streaming, which would fail without this.

https://github.com/apache/ozone/blob/030cf40fb1252f65de238e50ee13e6c0be58ef7e/hadoop-ozone/dist/src/main/smoketest/ozonefs/ozonefs.robot#L73-L77

Comment (kerneltime) [reply hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config:51]: That looks like a specific invocation to enable streaming vs. Setting it for all ozone ha tests.

Comment (adoroszlai) [reply hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config:51]: These are two separate settings.  `dfs.container.ratis.datastream.enabled=true` enables it at the server-side, but it's up to the client to decide if it uses regular or streaming path.  For FS API it is achieved by setting `ozone.fs.datastream.enabled`.  ([docs](https://ci-hadoop.apache.org/view/Hadoop%20Ozone/job/ozone-doc-master/lastSuccessfulBuild/artifact/hadoop-hdds/docs/public/feature/streaming-write-pipeline.html))

In short, if we want to run `ozonefs.robot` in some compose environment, we need to add this config.  It's just copied from the `ozone` and `ozonesecure` environments to the HA ones.

Comment (adoroszlai) [reply hadoop-ozone/dist/src/main/compose/ozone-ha/docker-config]: BTW, you are right, the config change is not needed for _this_ PR, rather the one this is based on (see description).  Now that the base PR is merged, I've updated this one, too, and now the change is reduced.  Sorry for the confusion.

**Review by smengcl — APPROVED** [review]
Nice improvement. Thanks @adoroszlai .

**Comment by adoroszlai** [comment]
Thanks @kerneltime, @smengcl for the review.
